apps/src/lib/levelbuilder/script-editor/ScriptEditor.jsx

```diff
@@ -241,6 +248,16 @@ class ScriptEditor extends React.Component {
           'Please provide a positive number of instructional minutes per week in Unit Calendar Settings.'
       });
       return;
+    } else if (
+      this.state.publishedState === 'Pilot' &&
+      this.state.pilotExperiment === ''
+    ) {
+      this.setState({
+        isSaving: false,
+        error:
+          'Please provide a pilot experiment in order to save with published state as pilot.'
+      });
```

Comment:
I would suggest extracting a variable `const shouldSave = this.state.publishedState === 'Pilot' && this.state.pilotExperiment === ''`, and using that here and below, instead of checking this.state below.